.ci/scripts/release.py

```diff
@@ -0,0 +1,156 @@
+        milestone_url = (
+            f"Redmine Milestone: {REDMINE_URL}/versions/{milestone_id}.json\n[noissue]"
+        )
+
+    print(f"\n\nRedmine stats: {json.dumps(stats, indent=2)}")
+    error_messages = []
+    if stats.get("status_not_modified"):
+        error_messages.append(
+            f"One or more issues are not MODIFIED {stats['status_not_modified']}"
+        )
+    if stats.get("without_milestone"):
+        error_messages.append(
+            f"One or more issues are not associated with a milestone {stats['without_milestone']}"
+        )
+    if len(project_set) > 1:
+        error_messages.append(f"Issues with different projects - {project_set}")
+    if error_messages:
+        error_messages.append(f"Verify at {redmine_query_url}")
+        raise RuntimeError("\n".join(error_messages))
+
+    return milestone_url
+
+
+print("===== Starting the release process =====")
+release_path = os.path.dirname(os.path.abspath(__file__))
+plugin_path = release_path.split("/.ci")[0]
+
+plugin_name = "pulp_installer"
+release_version = input("Please enter the pulp_installer version e.g. (3.12.0): ")
+pulpcore_version = input("Please enter the pulpcore version e.g. (3.12.0): ")
+pulpcore_selinux_version = input(
+    "Please enter the pulpcore-selinux version e.g. (1.2.4): "
+)
+
+issues_to_close = []
+for filename in Path(f"{plugin_path}/CHANGES").rglob("*"):
+    if filename.stem.isdigit():
+        issue = filename.stem
+        issue_url = f"{REDMINE_URL}/issues/{issue}.json"
+        issues_to_close.append(issue)
+
+issues = ",".join(issues_to_close)
+redmine_final_query = f"{REDMINE_QUERY_URL}{issues}"
+milestone_url = validate_redmine_data(redmine_final_query, issues_to_close)
+
+print("\n\nHave you checked the output of: $towncrier --version x.y.z --draft")
+print(f"\n\nRepo path: {plugin_path}")
+repo = Repo(plugin_path)
+git = repo.git
+
+git.checkout("HEAD", b=f"release_{release_version}")
+
+# First commit: changelog
+os.system(f"towncrier --yes --version {release_version}")
+git.add("docs/CHANGES.md")
+git.add("CHANGES/*")
+git.commit("-m", f"Building changelog for {release_version}\n\n[noissue]")
+
+
+# Second commit: release
+with open("roles/pulp_common/vars/main.yml") as f:
+    __pulp_version = yaml.safe_load(f)["__pulp_version"]
+    previous_pulpcore_version = __pulp_version.split('"')[1]
+
+with open("roles/pulp_common/defaults/main.yml") as f:
+    previous_selinux_version = yaml.safe_load(f)["__pulp_selinux_version"]
+
+with open("galaxy.yml") as f:
+    previous_installer_version = yaml.safe_load(f)["version"]
+
+sed(
+    [
+        "-i",
+        f"s/{previous_pulpcore_version}/{pulpcore_version}/",
+        "roles/pulp_common/vars/main.yml",
+    ]
+)
+sed(
+    [
+        "-i",
+        f"s/{previous_selinux_version}/{pulpcore_selinux_version}/",
+        "roles/pulp_common/defaults/main.yml",
+    ]
+)
+sed(["-i", f"s/{previous_installer_version}/{release_version}/", "galaxy.yml"])
```

Comment:
Do you see a way to make those replacements to be more specific? I know there is 
`sed /pattern/s/match_pattern/replace_pattern/` to only touch lines that match the first pattern.
(My concern is that we cannot assume that different projects will never have the same version by accident. Pulpcore e.g. just rolled over python versions.)

Reply:
I think it is possible, only the last sed needs to be general